Semantic changes

deekeh opened the pull request:
Added Semantic changes:
1. Changed wordings to make the parameter description more meaningful and easier to understand.
1. A quick search for the word "allowlist" suggests that "whitelist" is a better fit.

<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
Added semantic changes to improve English formatting.
<!-- ✍️ In a sentence or two, describe your changes -->

#### Motivation
<!-- ❓ Why are you making this change? Help us understand how your changes help readers. -->
When I came to know that `JSON.stringify()` has more than one parameter, I stumbled upon this page which describes the function perfectly, though I thought the sentence could be improved with respect to its formatting.

#### Supporting details
<!-- 🔗 Link to supporting information, such as bug trackers, source control, release notes, and vendor docs. -->
[Allowlist on Urban Dictionary](https://www.urbandictionary.com/define.php?term=Allowlist)

#### Related issues
<!-- 🔨 If applicable, use "Fixes #XYZ" -->
None

#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

deekeh committed:
Semantic changes

Added Semantic changes:
1. Changed wordings to make the parameter description more meaningful and easier to understand.
1. A quick search for the word "allowlist" suggests that "whitelist" is a better fit.

wbamberg reviewed (changes requested):
Thank you for your PR, @deekeh !

* changing "serve" to "serves" is definitely a fix for a grammatical error, so +1
* "allowlist" is intentional: https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Writing_style_guide#inclusive_language

Perhaps we could simplify the language here a bit? I'm not sure why we imply here (and in other parts of the page) that the values should be `String` or `Number` *objects*, when they can be just primitives. Could we say something like:

> A function that alters the behavior of the stringification process, or an array of strings or numbers that is used to filter the properties of the object that will be included in the JSON string.

What do you think?

deekeh committed:
Updating the description for parameter `replacer`

Updating the parameter description to be more precise and easy-to-understand to the user.

deekeh commented:
Thanks for the suggestion, @wbamberg!
Though I too, do not understand the reason behind using the term `object` when they can be just strings or numbers, I have not changed them and edited the description while retaining the term `object` over there. (Maybe it is referring to the funny behavior that *everything* in JavaScript is an object?!)

*Suggestion: I am also concerned that it not really described how the parameter accepts a `function`, that how the parameter `function` would look like. Maybe a simple example of parameter accepting a `function` would help too! I personally don't really get the `function` part of the parameter (as a lay-developer). So, I would also suggest adding an example of the same too.*

wbamberg commented on files/en-us/web/javascript/reference/global_objects/json/stringify/index.md:
```diff
@@ -34,10 +34,9 @@ JSON.stringify(value, replacer, space)
-    selecting/filtering the properties of the value object to be included in the JSON
-    string. If this value is {{JSxRef("null")}} or not provided, all properties of the
-    object are included in the resulting JSON string.
+    {{JSxRef("String")}}s or {{JSxRef("Number")}}s that will be filtered from the object
+    and included in the JSON string. If this value is {{JSxRef("null")}} or not provided,
```
```suggestion
    strings or numbers naming properties of `value` that should be included in the output. If this value is {{JSxRef("null")}} or not provided,
```

maybe?

deekeh committed:
Update files/en-us/web/javascript/reference/global_objects/json/stringify/index.md

Applying suggested changes

Co-authored-by: wbamberg <will@bootbonnet.ca>

wbamberg commented on files/en-us/web/javascript/reference/global_objects/json/stringify/index.md:
```diff
@@ -34,10 +34,8 @@ JSON.stringify(value, replacer, space)
-    selecting/filtering the properties of the value object to be included in the JSON
-    string. If this value is {{JSxRef("null")}} or not provided, all properties of the
-    object are included in the resulting JSON string.
+    strings or numbers naming properties of `value` that should be included in the output. If this value is {{JSxRef("null")}} or not provided,
```
```suggestion
    strings or numbers naming properties of `value` that should be included in the output. If `replacer` is {{JSxRef("null")}} or not provided,
```

wbamberg committed:
clarify "this value"

wbamberg reviewed (approved):
👍 thanks for this PR, @deekeh !
